### PR TITLE
Added some wallet utility methods getTotalReceived() and getTotalSent()

### PR DESCRIPTION
**Pull request by snakecharmer1024**
as well as a test for them.

In particular look at getTotalSent() and how it treats shared transactions (1 transaction signed by different wallets). I didn't add a test for that part yet, but everything else is checked.

Please let me know if there's anything else I should do.

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/core/Wallet.java [hunk 1]
```diff
@@ -3225,6 +3226,82 @@ private void checkBalanceFuturesLocked(@Nullable Coin avail) {
         }
     }
 
+    /**
+     * Returns the amount of bitcoin ever received via output. THIS IS NOT THE BALANCE! If an output spends from a
+     * transactions whose inputs are also to our wallet, the input amounts are deducted from the outputs contribution, with a minimum of zero
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/Wallet.java]: transactions→transaction

File: core/src/main/java/org/bitcoinj/core/Wallet.java [hunk 2]
```diff
@@ -3225,6 +3226,82 @@ private void checkBalanceFuturesLocked(@Nullable Coin avail) {
         }
     }
 
+    /**
+     * Returns the amount of bitcoin ever received via output. THIS IS NOT THE BALANCE! If an output spends from a
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/Wallet.java]: I don't think we need to scream this (uppercase letters). If you want to emphasize something, use HTML \<b\> or \<i\> tags.

**Comment by schildbach** [comment]
Thanks, this is a very nice contribution. I've annotated two nitpicks, but those are not blockers.

Btw. you can edit your commit and force-push it over your pull request. No need to close old PRs and open new ones.

**Comment by snakecharmer1024** [comment]
Great thanks! I'll try to figure out how to edit the commit and force-push.

**Comment by snakecharmer1024** [comment]
There I think I did it.

**Comment by schildbach** [comment]
Yeah, that worked. I merged your PR, thanks! Had to fix some formatting issues in the tests though. Please be careful about only using spaces (rather than tabs) for indentation.

**Comment by snakecharmer1024** [comment]
Will keep that in mind for next time! Thanks again

**Pull request closed without merging**